spec/arrays.dd

```diff
@@ -1008,10 +1008,11 @@ $(H3 $(LNAME2 strings, Strings))
 $(SPEC_RUNNABLE_EXAMPLE_FAIL
 ---------
 char[] str1 = "abc";                // error, "abc" is not mutable
```

Comment:
The bug report asks for a redesign of this example because currently the errors do not match. For example, the line above yields:

```
cannot implicitly convert expression `"abc"` of type `string` to `char[]`
```